### ISSUE TITLE
Load dalamud.dll without a file lock?

### ISSUE THREAD

**Issue by daemitus** [goatcorp/Dalamud#408]
Can the injector load dalamud.dll, without causing a file lock, while not completely screwing Assembly.Location?

**Comment by daemitus** [comment]
- attempt1: UnloadLibrary("Dalamud.dll")
 - CoreCLR error
- attempt2: `AppDomain.Unload(AppDomain.CurrentDomain)`
 - hard crash
- attempt3: Load Dalamud as a plugin
 - HarmonyLib does not work on collectible assemblies

**Comment by daemitus** [comment]
When re-injecting the same assembly: 
System.Exception: Service Dalamud.IoC.Internal.ServiceContainer was set twice

So we then need to zero out the IoC containers? This ultimately means that dalamud isnt unloading, the main thread is just terminating, but any statics are remaining.

tldr, dispose statics?
